### DISCUSSION TITLE
Pulse: <default server>: connection issue

### DISCUSSION THREAD

**Discussion by mackgm** [rtl-airband/RTLSDR-Airband#206] [Q&A]
I have installed RTLSDR-Airband according to the instructions [here](https://github.com/szpajder/RTLSDR-Airband/wiki).  When I start RTLSDR-Airband I get the messages below in my syslog.  
I am using the following commands to start RTLSDR-Airband:

sudo bash
pulseaudio --system --daemon
/usr/local/bin/rtl_airband -f -c /usr/local/etc/rtl_airband.conf

I have searched the internet but cannot find a solution.  How do I fix these errors?

Errors:
raspberrypi rtl_airband[1117]: pulse: <default_server>: connection failed: Access denied
raspberrypi rtl_airband[1117]: pulse: <default_server>: connecting...
raspberrypi pulseaudio[1106]: Denied access to client with invalid authentication data.

Sometimes I got these messages:
raspberrypi rtl_airband[987]: pulse: <default_server>: connecting...
raspberrypi rtl_airband[987]: pulse: <default_server>: connection failed: Connection refused
raspberrypi rtl_airband[987]: pulse: <default_server>: failed to connect: Invalid argument

**Comment by szpajder** [comment]
Questions and general discussions go into Discussions section. Please use Issues section only for reporting bugs in the program.

Have you set up your Pulseaudio server according to https://github.com/szpajder/RTLSDR-Airband/wiki/Configuring-PulseAudio-server ? Probably not, because it either isn't listening on the specified address:port ("Connection refused") or requires authentication ("Denied access...").

**Reply by mackgm** [reply]
Do I need to restart this thread in the Discussion session?

**Comment by szpajder** [comment]
Nope, I've already moved it there.

>

**Reply by mackgm** [reply]
Thanks!
I used the page above and I am still having problems.
I used these steps to install RTLSDR-Airband
    1  sudo apt-get install build-essential libmp3lame-dev libshout3-dev libconfig++-dev
    2  sudo apt-get install libraspberrypi-dev
    3  apt-get install librtlsdr-dev
    4  sudo apt-get install librtlsdr-dev
    5  sudo apt-get install libpulse-dev
    6  cd
    7  wget -O RTLSDR-Airband-3.2.1.tar.gz https://github.com/szpajder/RTLSDR-Airband/archive/v3.2.1.tar.gz
    8  tar xvfz RTLSDR-Airband-3.2.1.tar.gz
    9  cd RTLSDR-Airband-3.2.1
   10  make PLATFORM=rpiv2 PULSE=1
   11  sudo make install

When I type in the command "ps xa|grep pulseaudio" I get this - "1485 pts/0    S+     0:00 grep --color=auto pulseaudio"

When I start Airband as root it does not start Pulse.  Below is my Airband conf.
devices = (
 {
  type = "rtlsdr";
  index = 0;
  gain = 25;
  mode = "scan";
  centerfreq = 120.0;
  correction = 1;
  channels = (
      {
         freqs = ( 123.05 );
         squelch = 2;
         outputs = (
             {
             type = "pulse";
             }
             );
       });
});

What else can I check or do?
Thanks!

**Comment by szpajder** [comment]
rtl_airband won't start your Pulseaudio server, you have to do it yourself. Usually this is done automatically when you start your desktop environment like KDE or Gnome or whatever.

You didn't specify your PA server address in the pulse output configuration section, which causes the program to connect to the default server, which is localhost. Is it where your PA server is supposed to be running?

**Reply by mackgm** [reply]
I am not using a desktop.  I am trying to set this up as a headless system.

I don't have a server address because I want the pi that is running rtl_airband to be the server to itself.  

I have set the system up using all the instructions from Dave at LiveATC and it still will not play any sound out of the speakers even when doing a speaker test from the command prompt.

I must be missing something.

**Reply by feelingwalnut** [reply]
Ive tried to set up similar and have yet to do so sucessfully.
But it sounds likeyour server address will be localhost or the appropriate 127.0.0.1

**Comment by szpajder** [comment]
OK, so you want to run Pulseaudio in system mode. In this case start here:

https://www.freedesktop.org/wiki/Software/PulseAudio/Documentation/User/SystemWide/
https://unix.stackexchange.com/questions/75404/user-systemd-pulseaudio-service
https://possiblelossofprecision.net/?p=1956

**Reply by mackgm** [reply]
I have decided to not run pulseaudio as a system wide service.  I am going to install a gui to use the local audio.  What program would you recommend to use with the gui so I can hear the output of RTLSDR_Airband?

**Reply by feelingwalnut** [reply]
are you trying to listen to multiple frequencies? 
rtl_fm will be much simpler for single frequency or "scanning" through a group of known frequencies and outputting audio locally. write a script to act as your "config" and you need only click to launch your frequencies.
airband is best for recording or broadcasting multiple transmissions that are simultaneous.

lightest "gui" i've used that is quite functional is JWM, next is lxde manually installed, not through metas, for as little bloat as possible.

**Comment by mackgm** [comment]
Thanks for the feedback.  I am only using one frequency and I
recently changed over to rtl_fm and it is easier.

**Reply by feelingwalnut** [reply]
Airband is excellent, especially for my purpose of archiving multiple freq and then acessing remotely.

But i'd a feeling you might have found rtl_fm more to your needs.

I dont have it in front of me but a simple script would be on your desktop, have the shebang and a single line of rtl_fm code with desired freq, gain and such or multitude of freqs or scan range.
Give it permissions and have it running with a click.
I use airband for my base and the others like rtl_power and rtl_fm for the field.

**Comment by ckarrie** [comment]
Hey all,
i'm running into the same problem:

- Raspberry PI 3
- 2 working RTL-SDR dongles (one of them used by PiAware, second should be used by Airband)
- compiled RTLSDR-Airband with `make PLATFORM=rpiv2 PULSE=1 NFM=1`
 - running without desktop (headless raspbian)

Pulseaudio-Config:

```
...
load-module module-native-protocol-tcp auth-ip-acl=127.0.0.1;192.168.178.0/24 auth-anonymous=1
load-module module-zeroconf-publish
...
```

RTLSDR-Airband config:

```
devices:
({
  type = "rtlsdr";
  index = 1;
  gain = 100;
  centerfreq = 173.0;
  correction = 0;
  mode = "multichannel";
  channels:
  (
    {
      freq = 172.640;
      modulation = "nfm";
      outputs: ({
          type = "pulse";
          #server = "127.0.0.1";
          sink = "chan0";
          stream_name = "172.64Mhz";
          continuous = true;
      });
    },
    {
      freq = 172.680;
      modulation = "nfm";
      outputs: ({
          type = "pulse";
          #server = "127.0.0.1";
          sink = "chan1";
          stream_name = "172.68Mhz";
          continuous = true;
      });
    },
    {
      freq = 172.781;
      modulation = "nfm";
      outputs: ({
          type = "pulse";
          #server = "127.0.0.1";
          sink = "chan2";
          stream_name = "chan2";
          continuous = true;
      });
    },
    {
      freq = 173.1395;
      modulation = "nfm";
      outputs: ({
          type = "pulse";
          #server = "127.0.0.1";
          sink = "chan3";
          stream_name = "chan3";
          continuous = true;
      });
    },
    {
      freq = 173.1696;
      modulation = "nfm";
      outputs: ({
          type = "pulse";
          #server = "127.0.0.1";
          sink = "chan4";
          stream_name = "chan4";
          continuous = true;
      });
    },
    {
      freq = 173.20;
      modulation = "nfm";
      outputs: ({
          type = "pulse";
          #server = "127.0.0.1";
          sink = "chan5";
          stream_name = "chan5";
          continuous = true;
      });
    }
  );
 }
);
```

Errors in syslog:

```
Jan 31 14:05:16 radiopi rtl_airband[1091]: pulse: <default_server>: connecting...
Jan 31 14:05:16 radiopi rtl_airband[1091]: pulse: <default_server>: connection failed: Connection refused
Jan 31 14:05:16 radiopi rtl_airband[1091]: pulse: <default_server>: failed to connect: Invalid argument
```

**Comment by ckarrie** [comment]
If I start pulseaudio with --system, I get following warnings:

`W: [pulseaudio] protocol-native.c: Denied access to client with invalid authentication data.`

**Reply by ckarrie** [reply]
Verbose output of `sudo pulseaudio --system -vvvv`

```
I: [pulseaudio] client.c: Freed 4 "Native client (UNIX socket client)"
I: [pulseaudio] protocol-native.c: Connection died.
I: [pulseaudio] client.c: Freed 3 "Native client (UNIX socket client)"
I: [pulseaudio] protocol-native.c: Connection died.
I: [pulseaudio] client.c: Freed 2 "Native client (UNIX socket client)"
I: [pulseaudio] protocol-native.c: Connection died.
I: [pulseaudio] client.c: Freed 1 "Native client (UNIX socket client)"
I: [pulseaudio] protocol-native.c: Connection died.
I: [pulseaudio] client.c: Freed 0 "Native client (UNIX socket client)"
I: [pulseaudio] protocol-native.c: Connection died.
I: [pulseaudio] client.c: Freed 5 "Native client (UNIX socket client)"
I: [pulseaudio] protocol-native.c: Connection died.
I: [pulseaudio] client.c: Created 6 "Native client (UNIX socket client)"
D: [pulseaudio] protocol-native.c: Protocol version: remote 32, local 32
I: [pulseaudio] protocol-native.c: Got credentials: uid=0 gid=0 success=0
W: [pulseaudio] protocol-native.c: Denied access to client with invalid authentication data.
I: [pulseaudio] client.c: Created 7 "Native client (UNIX socket client)"
I: [pulseaudio] client.c: Freed 6 "Native client (UNIX socket client)"
I: [pulseaudio] protocol-native.c: Connection died.
D: [pulseaudio] protocol-native.c: Protocol version: remote 32, local 32
I: [pulseaudio] protocol-native.c: Got credentials: uid=0 gid=0 success=0
W: [pulseaudio] protocol-native.c: Denied access to client with invalid authentication data.
I: [pulseaudio] client.c: Created 8 "Native client (UNIX socket client)"
I: [pulseaudio] client.c: Freed 7 "Native client (UNIX socket client)"
I: [pulseaudio] protocol-native.c: Connection died.
D: [pulseaudio] protocol-native.c: Protocol version: remote 32, local 32
I: [pulseaudio] protocol-native.c: Got credentials: uid=0 gid=0 success=0
W: [pulseaudio] protocol-native.c: Denied access to client with invalid authentication data.
```

**Comment by ckarrie** [comment]
Update:
- pulseaudio running as user without errors now, 
  - removed `load-module module-zeroconf-publish` and 
  - removed `load-module module-native-protocol-unix auth-anonymous=1` from `/etc/pulse/default.pa`, 
  - added `default-server = 127.0.0.1` to `/etc/pulse/client.conf`

but still not my sinks in `pacmd list-sinks`

It seems that the clients continues to disconnect:

```
I: [pulseaudio] client.c: Created 3 "Native client (TCP/IP client from 127.0.0.1:38970)"
I: [pulseaudio] protocol-native.c: Client authenticated anonymously.
I: [pulseaudio] socket-server.c: TCP connection accepted by tcpwrap.
I: [pulseaudio] client.c: Created 4 "Native client (TCP/IP client from 127.0.0.1:38972)"
I: [pulseaudio] protocol-native.c: Client authenticated anonymously.
D: [pulseaudio] protocol-native.c: Protocol version: remote 32, local 32
D: [pulseaudio] protocol-native.c: SHM possible: no
D: [pulseaudio] protocol-native.c: Negotiated SHM: no
D: [pulseaudio] protocol-native.c: Disabling srbchannel, reason: No SHM support
D: [pulseaudio] protocol-native.c: Protocol version: remote 32, local 32
D: [pulseaudio] protocol-native.c: SHM possible: no
D: [pulseaudio] protocol-native.c: Negotiated SHM: no
D: [pulseaudio] protocol-native.c: Disabling srbchannel, reason: No SHM support
D: [pulseaudio] module-augment-properties.c: Looking for .desktop file for rtl_airband
D: [pulseaudio] module-augment-properties.c: Looking for .desktop file for rtl_airband
I: [pulseaudio] client.c: Freed 4 "rtl_airband"
I: [pulseaudio] protocol-native.c: Connection died.
I: [pulseaudio] client.c: Freed 3 "rtl_airband"
I: [pulseaudio] protocol-native.c: Connection died.
```

**Comment by ckarrie** [comment]
Hi, I would love to get some help/ideas here :-). This actually blocks me from using RTLSDR-Airband with BW3.
